src/app/components/core/backup-restore/backup-restore.component.ts

```diff
@@ -191,16 +191,24 @@ export class BackupRestoreComponent implements OnInit, OnDestroy {
   }
 
   public async downloadBackup(backup): Promise<void> {
-    const blob = await this.backupRestoreService.downloadBackup(backup.id);
-    const url = window.URL.createObjectURL(blob);
-    // create a custom anchor tag
-    const a = document.createElement('a');
-    a.href = url;
-    const date = this.dateFormatter.transform(backup.date, 'YYYY_MM_DD_HH_mm_ss');
-    a.download = 'fledge_backup_' + date + '.tar.gz';
-    document.body.appendChild(a);
-    a.click();
-    document.body.removeChild(a);
+    try {
+      const blob = await this.backupRestoreService.downloadBackup(backup.id);
+      const url = window.URL.createObjectURL(blob);
+      // create a custom anchor tag
+      const a = document.createElement('a');
+      a.href = url;
+      const date = this.dateFormatter.transform(backup.date, 'YYYY_MM_DD_HH_mm_ss');
+      a.download = 'fledge_backup_' + date + '.tar.gz';
+      document.body.appendChild(a);
+      a.click();
+      document.body.removeChild(a);
+    } catch (error) {
+      if (error.status === 0) {
+        console.log('service down ', error);
+      } else {
+        this.alertService.error(error.statusText);
```

Comment:
gradually the API is being improved to return JSON response body for error 4xx/5xx cases, not only the  status text (reason) in header. We should handle this gracefully and give priority to response error message and fallback to statusText.